subsys/suit/mci/src/suit_mci_nrf9280.c

```diff
@@ -0,0 +1,768 @@
+
+	if (class_id == NULL || policy == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	if (suit_storage_mpi_get(class_id, &mpi) != SUIT_PLAT_SUCCESS) {
+		return MCI_ERR_MANIFESTCLASSID;
+	}
+
+	if (suit_storage_mpi_role_get(class_id, &role) != SUIT_PLAT_SUCCESS) {
+		return MCI_ERR_MANIFESTCLASSID;
+	}
+
+	*policy = suit_mpi_independent_updateability_policy_to_metadata(
+		mpi->independent_updateability_policy);
+	if (*policy == SUIT_INDEPENDENT_UPDATE_UNKNOWN) {
+		return SUIT_PLAT_ERR_OUT_OF_BOUNDS;
+	}
+
+	/* Override independent updateability policy in recovery scenarios:
+	 * If the update candidate was delivered by the recovery firmware,
+	 * it must not update the recovery firmware.
+	 * Invoke modes included, so the recovery firmware may reject the incorrect
+	 * update candidate before resetting the SoC.
+	 */
+	switch (suit_execution_mode_get()) {
+	case EXECUTION_MODE_INVOKE_FOREGROUND_DFU:
+	case EXECUTION_MODE_INVOKE_RECOVERY:
+	case EXECUTION_MODE_INSTALL_FOREGROUND_DFU:
+	case EXECUTION_MODE_INSTALL_RECOVERY:
+	case EXECUTION_MODE_POST_INVOKE_FOREGROUND_DFU:
+	case EXECUTION_MODE_POST_INVOKE_RECOVERY:
+		if ((role == SUIT_MANIFEST_APP_RECOVERY) || (role == SUIT_MANIFEST_RAD_RECOVERY)) {
+			*policy = SUIT_INDEPENDENT_UPDATE_DENIED;
+		}
+		break;
+
+	case EXECUTION_MODE_FAIL_INSTALL_NORDIC_TOP:
+		/* In this state, only the Nordic top manifest is accepted as an update candidate.*/
+		if (role != SUIT_MANIFEST_SEC_TOP) {
+			*policy = SUIT_INDEPENDENT_UPDATE_DENIED;
+		}
+		break;
+
+	default:
+		break;
+	}
+
+	return SUIT_PLAT_SUCCESS;
+}
+
+mci_err_t suit_mci_manifest_class_id_validate(const suit_manifest_class_id_t *class_id)
+{
+	if (class_id == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	suit_manifest_role_t role = SUIT_MANIFEST_UNKNOWN;
+	suit_plat_err_t ret = suit_storage_mpi_role_get(class_id, &role);
+
+	if (ret != SUIT_PLAT_SUCCESS) {
+		return MCI_ERR_MANIFESTCLASSID;
+	}
+
+	return SUIT_PLAT_SUCCESS;
+}
+
+static bool skip_validation(suit_manifest_role_t role)
+{
+#ifdef CONFIG_SDFW_LCS
+	/* Read the domain-specific LCS value. */
+	enum lcs current_lcs = LCS_DISCARDED;
+
+	switch (role) {
+	case SUIT_MANIFEST_SEC_TOP:
+	case SUIT_MANIFEST_SEC_SDFW:
+	case SUIT_MANIFEST_SEC_SYSCTRL:
+		return false;
+
+	case SUIT_MANIFEST_APP_ROOT:
+	case SUIT_MANIFEST_APP_RECOVERY:
+	case SUIT_MANIFEST_APP_LOCAL_1:
+	case SUIT_MANIFEST_APP_LOCAL_2:
+	case SUIT_MANIFEST_APP_LOCAL_3:
+		current_lcs = lcs_get(LCS_DOMAIN_ID_APPLICATION);
+		break;
+
+	case SUIT_MANIFEST_RAD_RECOVERY:
+	case SUIT_MANIFEST_RAD_LOCAL_1:
+	case SUIT_MANIFEST_RAD_LOCAL_2:
+		current_lcs = lcs_get(LCS_DOMAIN_ID_RADIOCORE);
+		break;
+
+	default:
+		return false;
+	}
+
+	if ((current_lcs == LCS_ROT) || (current_lcs == LCS_ROT_DEBUG) ||
+	    (current_lcs == LCS_EMPTY)) {
+		return true;
+	}
+#endif /* CONFIG_SDFW_LCS */
+
+	return false;
+}
+
+mci_err_t suit_mci_signing_key_id_validate(const suit_manifest_class_id_t *class_id,
```

Comment:
There is an upcoming change to this function (it will be renamed + little change in functionality in here): https://github.com/nrfconnect/sdk-nrf/pull/20050/files#diff-5f831b3a82f1e14d189946f985f443439e1f68e17d67551dc146b0f47735fe18R161 - depending on which PR will come first we will have to align this in one PR or the other